### PR TITLE
✨(kustomize/v1, go/v3) : Add new LeaderElectionReleaseOnCancel as true option commented to let users know when they can use it

### LINKED ISSUE #2593: Set `LeaderElectionReleaseOnCancel` to the scaffolded main.go
### What do you want to happen?

Kubebuilder is generating `main.go` scripts that are getting closed immediately when the `Manager` is done.
Given that, we can add `LeaderElectionReleaseOnCancel: true` to the manager (with an explanation comment) to improve the performance of re-election.

### Extra Labels

_No response_

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
Fixes #2593

### CODE REVIEW

**Comment by NikhilSharmaWe** [comment]
Error logs says `vet: ./main.go:64:3: unknown field LeaderElectionReleaseOnCancel in struct literal`. Unable to understand the cause of it. The function [`NewManager`](https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/main.go#L68) uses type `Options` as one of the parameters, since Options is actually `[manager.Options](https://github.com/kubernetes-sigs/controller-runtime/blob/master/alias.go#L49)` and since it has [`LeaderElectionReleaseOnCancel`](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/manager.go#L176) type in it why this error is being displayed.

**Comment by AlmogBaku** [comment]
This feature wasn't supported in the `controller-runtime:v0.6.4` which is the version that the go/v2 plugin uses - therefore, you should only patch go/v3

File: pkg/plugins/golang/v3/scaffolds/internal/templates/main.go [hunk 1]
```diff
@@ -238,13 +238,16 @@ func main() {
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
 {{ if not .ComponentConfig }}
+	// Setting LeaderElectionReleaseOnCancel to true prevents main.go 
+	// scripts to close when Manager is done and improves the performance of re-election.
```
Comment (NikhilSharmaWe) [inline pkg/plugins/golang/v3/scaffolds/internal/templates/main.go:242]: @AlmogBaku All tests are passing now. Could you please review this explanation comment added before moving forward.

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
I think we need to discuss if this option should or NOT be added by default?
Also, what is the motivation for that?
What issue are you facing that you want to add it in the default scaffold.

See : https://github.com/kubernetes-sigs/kubebuilder/issues/2593#issuecomment-1087236601

/hold

**Comment by NikhilSharmaWe** [comment]
@camilamacedo86 and @AlmogBaku Do we also need to add the comment here also https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3-config/main.go#L66? How can we check that it is possible via the ComponentConfig as well (since @camilamacedo86 mentioned that this is needed for this case)?

**Comment by camilamacedo86** [comment]
Hi @NikhilSharmaWe, 

I think we also need to update the template used by the ComponentConfig option so that we will check its results in the sample. Why? Ensure that all have the same options, and follow the same standard making it easier for us to keep the project maintained.

**Comment by NikhilSharmaWe** [comment]
@camilamacedo86 Are talking about adding the comment for this case as well?
https://github.com/kubernetes-sigs/kubebuilder/blob/43525de6224aa6381acf2733411ca383f46c26ce/pkg/plugins/golang/v3/scaffolds/internal/templates/main.go#L262

**Comment by camilamacedo86** [comment]
Hi @NikhilSharmaWe, 

The change shows to be here: https://github.com/kubernetes-sigs/kubebuilder/blob/019c1854f67e3bd61af5aa834a1573e6934b2e8b/pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/manager/controller_manager_config.go#L47-L57

We need to pass the option under the leader election options
That should be also commented ( since it is an yaml file to comment you will use the prefix # )
Also, it should have the same comment explanation.

**Review by camilamacedo86 — APPROVED** [review]
Great work 🥇 

/hold cancel

**Review by everettraven — APPROVED** [review]
/lgtm

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

**Pull request merged**